src/QtAVPlayer/qavdemuxer.cpp

```diff
@@ -450,27 +451,33 @@ int QAVDemuxer::resetCodecs()
 {
     Q_D(QAVDemuxer);
     int ret = 0;
+
+    QAVDictionaryHolder opts;
+    for (const auto & key: d->codecOptions.keys())
+        av_dict_set(&opts.dict, key.toUtf8().constData(), d->codecOptions[key].toUtf8().constData(), 0);
+
     for (std::size_t i = 0; i < d->ctx->nb_streams && ret >= 0; ++i) {
         if (!d->ctx->streams[i]->codecpar) {
             qWarning() << "Could not find codecpar";
             return AVERROR(EINVAL);
         }
+
         const AVMediaType type = d->ctx->streams[i]->codecpar->codec_type;
         switch (type) {
             case AVMEDIA_TYPE_VIDEO:
             {
                 QSharedPointer<QAVCodec> codec(new QAVVideoCodec);
                 d->availableStreams.push_back({ int(i), d->ctx, codec });
-                ret = setup_video_codec(d->inputVideoCodec, d->ctx->streams[i], *static_cast<QAVVideoCodec *>(codec.data()));
+                ret = setup_video_codec(d->inputVideoCodec, d->ctx->streams[i], *static_cast<QAVVideoCodec *>(codec.data()), &opts.dict);
             } break;
             case AVMEDIA_TYPE_AUDIO:
                 d->availableStreams.push_back({ int(i), d->ctx, QSharedPointer<QAVCodec>(new QAVAudioCodec) });
-                if (!d->availableStreams.last().codec()->open(d->ctx->streams[i]))
+                if (!d->availableStreams.last().codec()->open(d->ctx->streams[i], &opts.dict))
                     qWarning() << "Could not open audio codec for stream:" << i;
                 break;
             case AVMEDIA_TYPE_SUBTITLE:
                 d->availableStreams.push_back({ int(i), d->ctx, QSharedPointer<QAVCodec>(new QAVSubtitleCodec) });
-                if (!d->availableStreams.last().codec()->open(d->ctx->streams[i]))
+                if (!d->availableStreams.last().codec()->open(d->ctx->streams[i], &opts.dict))
```

Comment:
a bit suspicious why subtitle codec would need something from video codec, is it possible that they share the same options?

Reply:
it smells it would need to rename the opts to `videoCodecOptions` and use it only for video codec? in `setup_video_codec` and keep empty opts for audio and subtitle, or add later audioCodecOptions() if needed ?

Reply:
I completely agree with you, it has been fixed  (left for now only for the video)